src/modules/general/slash-commands/set-profile.command.ts

```diff
@@ -0,0 +1,170 @@
+import { BaseCommand } from "@abstracts/command/command.abstract";
+import { CmdArg, CmdType } from "@abstracts/command/command.types";
+import { SlashCommand } from "@decorators/commands.decorator";
+import { ProfilePresetEntity, UserEntity } from "@entities/index";
+import {
+  NavigationButtons,
+  getNavigationSetup,
+} from "@helpers/navigation.helper";
+import PaginationHelper from "@helpers/pagination.helper";
+import {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  ComponentType,
+  EmbedBuilder,
+  GuildMember,
+} from "discord.js";
+
+import { defaultTemplate } from "./profile.command";
+import { gatherProfileTokens } from "@utils/gather-tokens.util";
+
+const TO_PROCEED = "to-proceed-button";
+
+@SlashCommand({
+  name: "set-profile",
+  description: "sets your profile",
+})
+export class SetProfileCommand extends BaseCommand<CmdType.SLASH_COMMAND> {
+  async execute(interaction: CmdArg<CmdType.SLASH_COMMAND>) {
+    await interaction.deferReply();
+
+    const userData = await UserEntity.findOne({
+      where: {
+        uid: interaction.user.id,
+      },
+      relations: {
+        inventory: true,
+        wallet: true,
+        profile_presets: true,
+        selected_preset: true,
+      },
+    }).then((user) => {
+      if (!user) {
+        return UserEntity.save({
+          uid: interaction.user.id,
+          inventory: {},
+          wallet: {},
+        });
+      } else {
+        return user;
+      }
+    });
+
+    if (!userData.profile_presets || userData.profile_presets?.length == 0) {
+      return this.sendError(
+        "You don't have any profile presets, please create one",
+        interaction
+      );
+    }
+
+    // Creating pseudo preset list, to not mess with the original data
+    const presets = [
+      {
+        id: "default",
+        json: JSON.stringify(defaultTemplate),
+      } as ProfilePresetEntity,
+      ...userData.profile_presets,
+    ];
```

Comment:
you also want to have all presets that doesn't have user relations. those presets are created by admins with /profile-presets command. but if we want to sell them later with currency your current variant is good. in the future we can just assign admin's presets to user and then he will get them here

Reply:
> you also want to have all presets that doesn't have user relations. those presets are created by admins with /profile-presets command. but if we want to sell them later with currency your current variant is good. in the future we can just assign admin's presets to user and then he will get them here

I think that selling presets is our best variant. As idea, we can have public shop with all presets made by users, and sell them with lower price that creating a new one. We also can send % of price to original creator. I will leave it as it is, and edit by needs after we will have the shop.